dask/utils.py

```diff
@@ -1229,6 +1229,16 @@ def get_scheduler_lock(collection=None, scheduler=None):
 
     if actual_get == multiprocessing.get:
         return multiprocessing.get_context().Manager().Lock()
+    else:
+        # if this is a distributed client with processes, we also need to lock
+        try:
+            from distributed.worker import get_client
+
+            client = get_client()
```

Comment:
might be nicer to avoid calling `get_worker(` using:

```python
try:
    from distributed import Client
    from distributed.utils import get_mp_context
except ImportError:
    pass
else:
    if isinstance(get.__self__, Client):
        return get_mp_context().Manager.Lock()
```